src/ipc/ipc4/helper.c

```diff
@@ -673,7 +673,7 @@ int ipc4_trigger_chain_dma(struct ipc *ipc, struct ipc4_chain_dma *cdma)
 	struct ipc_comp_dev *ipc_pipe;
 	struct comp_dev *host;
 	uint32_t pipeline_id;
-	int ret;
+	int ret = 0;
```

Comment:
why would the compiler complain here? I really don't see a single non-obvious path

Reply:
Seppo had already fixed the issue (much better than I did) with commit 6c48d42f806b77d5d63cb3ded24b7ab40ade7973

I'll drop the last commit.